Adding swagger json and examples for Compute Diagnostic Resource

ansahdev opened the pull request:
Provider - preview

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [x] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

ansahdev committed:
Adding swagger json and examples for Compute Diagnostic Resource
Provider - preview

ansahdev committed:
Adding swagger json and examples for Compute Diagnostic Resource
Provider - preview

ansahdev committed:
Adding swagger json and examples for Compute Diagnostic Resource
Provider - preview

ansahdev committed:
Updated readme.md

ansahdev committed:
Updated readme.md

ansahdev committed:
Updated readme.md

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,247 @@
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/DiagnosticOperations/{operationId}": {
```
The resourceType should follow the lowerCamelCase convention and be plural. Please specify name as 'diagnosticOperations'.
Refer https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#arguments-for-crud-on-resource

ansahdev replied:
Updated.

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,247 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
```
Does this API return any response?

ansahdev replied:
No, nothing in the response body.

ansahdev replied:
No response content in body.

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,247 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/Diagnostics/runDiskInspection": {
```
POST must not be used on a collection of resources. POST should be used on a single resource like below
/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/{diagnosticId}/runDiskInspection

Refer best practice 'POST on collection segments' in https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html

waynekuo replied:
Thanks, Ramyasree! We are aware of that for Azure 1P RPs, POST is usually an operational task on an existing resource and PUT is the initial create operation (i.e. PUT VM). And this is keeping in mind that PUT and POST have always been somewhat ambiguous even outside the context of Azure, i.e. "https://restfulapi.net/rest-put-vs-post/". The RFC definition seems somewhat different from the way Azure is interpreting this too: https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.6.  

For our reasoning, we had thought that the Azure interpretation of POST vs PUT would **only** apply to ARM tracked resource types and we didn't know it would apply for Proxy-Only resource types such as ours as well. (It _is_ a bit confusing since I think we can probably find a few other existing Proxy-Only resource types that accept POST without a singular resource Id reference, but we don't have to digress and go down that path :) ) For our resource type, since each of our diagnostic run is a discrete operational task on its own where no ARM tracked resource is created, we had opted for the POST verb with no tracking unique identifier.

If you feel strongly that the same POST/PUT interpretation should apply to Proxy-Only resource types as well, I think we would prefer to replace the verb with PUT from POST. Asking the caller or client to generate a unique identifier for a diagnostic run is a bigger change and it makes the call feel a lot more like an ARM tracked resource. 

Please let us know what you think.

RamyasreeChakka replied:
POST guidance applies to proxy resources as well. Existing APIs that didn’t follow this guidance must have been introduced long time back. Lately ARM API review board is enforcing the guidelines strictly.

PUT doesn’t make sense in your scenario as runDiskInspection is an action. POST is the right verb to use. We can explore modeling the API something like below. Please setup a short meeting with me tomorrow to close on this.

POST /subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection/run

ansahdev replied:
I have updated swagger as per discussion -
1) Updated API for POST call
2) Added GET calls for Diagnostic collection and DiskInspection type.

RamyasreeChakka replied:
@ansahdev - I have added some comments. Once you resolve them, I will sign off.

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,247 @@
+          }
+        ],
+        "responses": {
+          "202": {
```
As this API is a long running operation, add autorest extension x-ms-long-running-operation and x-ms-long-running-operation-options
Refer http://azure.github.io/autorest/extensions/#x-ms-long-running-operation

ansahdev replied:
Added.

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,247 @@
+        "manifest": {
+          "type": "string"
+        },
+        "uploadSasUri": {
```
As sas uri is a secret, specify "x-ms-secret": true

ansahdev replied:
Added.

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,247 @@
+    },
+    "RunDiskInspectionAsyncOperationResult": {
+      "properties": {
+        "Result": {
```
Property names should be defined in lowerCamelCase. Comment applicable to other places too.

ansahdev replied:
Updated.

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/examples/GetDiagnosticOperation.json:
```diff
@@ -0,0 +1,24 @@
+    "responses": {
+        "200": {
+            "headers": {
+                "azure-asyncoperation": "Succeeded",
```
Azure-AsyncOperation header should be a absolute URI.
Please refer rpc async guidelines here: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md

ansahdev replied:
This operation outputs the final-state or the polling url via "location" header. As per http://azure.github.io/autorest/extensions/#x-ms-long-running-operation, the value could be one of "azure-async-operation" or "location" or "original-uri". Will remove the additional header - "azure-async-operation". Please advise.

RamyasreeChakka replied:
sounds good.

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,247 @@
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/DiagnosticOperations/{operationId}": {
+      "get": {
```
Can you specify examples for these APIs using x-ms-examples?
These examples will be used in the public documentation.

ansahdev replied:
Added.

ansahdev committed:
Resolved PR comments.

ansahdev committed:
Resolved PR comments.

ansahdev committed:
Resolved PR comments.

RamyasreeChakka commented:
As discussed offline, please implement below APIs

POST /subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection/run

GET /subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection
{
    id: xxx,
	name: diskInspection,
	properties: {
		supportedResourceTypes: [ "VMs"]		
	}
}

GET /subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics

{
  value: [
	{
		id: xxx,
		name: diskInspection,
		properties: {
			supportedResourceTypes: [ "VMs"]
		}
	}
  ]
}

ansahdev committed:
Updated swagger json with new APIs and updated examples based on PR comments.

ansahdev committed:
Updated swagger json with new APIs and updated examples based on PR comments.

ansahdev committed:
Updated swagger json with new APIs and updated examples based on PR comments.

ansahdev committed:
Added missing required property "name" in RunDiskInspectionAsyncOperationResult, x-ms-enum property.

ansahdev committed:
Added missing required property "name" in RunDiskInspectionAsyncOperationResult, x-ms-enum property.

ansahdev committed:
Added missing required property "name" in RunDiskInspectionAsyncOperationResult, x-ms-enum property.

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,367 @@
+            "type": "string"
+          },
+          "description": "Describes what are the supported resource types for a diagnostic",
+          "x-ms-examples": [ "VMs" ]
```
Update the example value  VMs to "Microsoft.Compute/virtualMachines"

ansahdev replied:
Updated.

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,367 @@
+        },
+        "virtualMachineName": {
+          "description": "Virtual machine name.",
+          "type": "string"
```
Use generic property name resourceId or resource name  so that you can extend disk inspection functionality to other resources in compute in future.

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,367 @@
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics": {
+      "get": {
+        "tags": [
```
As this is a get collection, follow the best practice of defining x-ms-pagaeable & nextLink property even though you have only singleton object. In future, if your API becomes pageable, you don't need to increment the api version.
[http://azure.github.io/autorest/extensions/#x-ms-pageable](http://azure.github.io/autorest/extensions/#x-ms-pageable)

ansahdev replied:
Added "x-ms-pageable": {
          "nextLinkName":  null   //setting to null as "nextlink" property is not available.
        },
as per the documentation

RamyasreeChakka replied:
Instead of null, use nextLink. In response, you can set the nextLink value to null.

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/examples/GetDiagnosticOperation.json:
```diff
@@ -0,0 +1,22 @@
+      "headers": {
+        "Retry-After": "120000",
+        "x-ms-request-id": "57b891ab-1fb7-4f5a-b002-71eb6854961f",
+        "location": "https://westcentralus.diagnostic.compute.azure.com/subscriptions/88fd8cb2-8248-499e-9a2d-4929a4b0133c/providers/Microsoft.Compute/locations/southcentralus/diagnosticOperations57b891ab-1fb7-4f5a-b002-71eb6854961f"
```
>location [](http://example.com/codeflow?start=9&length=8)

nit: '/' is missing after diagnosticOperations

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,367 @@
+    "ComputeDiagnosticBase": {
+      "description": "Contains metadata of a diagnostic type",
+      "properties": {
+        "id": {
```
>id [](http://example.com/codeflow?start=9&length=2)

Top level properties 'id', 'name', 'type' are must defined. Add type property.

ansahdev replied:
Added  "type": "object"

RamyasreeChakka commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/examples/GetDiagnostics.json:
```diff
@@ -0,0 +1,29 @@
+            {
+              "id": "diskInspection",
+              "name": "Disk Inspection",
+              "properties": {
```
Add type property too. 
Value would be Microsoft.Compute/locations/diagnostics

ansahdev committed:
PR comments -
1) Added "x-ms-pageable" property on get
2) Updated "POST" call input model
3) Added "type" property in input model.
4) Fixed typo.

ansahdev committed:
PR comments -
1) Added "x-ms-pageable" property on get
2) Updated "POST" call input model
3) Added "type" property in input model.
4) Fixed typo.

ansahdev committed:
PR comments -
1) Added "x-ms-pageable" property on get
2) Updated "POST" call input model
3) Added "type" property in input model.
4) Fixed typo.

ansahdev committed:
Fixed typo in diagnostic.json

ansahdev committed:
Fixed typo in diagnostic.json

ansahdev committed:
Fixed typo in diagnostic.json

RamyasreeChakka reviewed (approved):
:shipit:

ansahdev committed:
Added "nextLink" property in GetDiagnostics response, setting it to null as the response is a not paged.

ansahdev committed:
Added "nextLink" property in GetDiagnostics response, setting it to null as the response is a not paged.

ansahdev committed:
Added "nextLink" property in GetDiagnostics response, setting it to null as the response is a not paged.

ansahdev committed:
Fixed spacing
Added missing comma in diagnostic.json

ansahdev committed:
Fixed spacing
Added missing comma in diagnostic.json

ansahdev committed:
Fixed spacing
Added missing comma in diagnostic.json

ansahdev committed:
Prettier check fix

ansahdev committed:
Prettier check fix

ansahdev committed:
Prettier check fix

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/examples/GetDiagnostics.json:
```diff
@@ -0,0 +1,31 @@
+        "x-ms-request-id": "27b7c568-16ec-46f3-bcf1-5bea3f2529b1"
+      },
+      "body": {
+        "properties": {
+          "description": "Gets information of all compute diagnostic types.",
```
We should not have these two levels
```suggestion
```

ansahdev replied:
Updated

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/examples/GetDiagnostics.json:
```diff
@@ -0,0 +1,31 @@
+          ],
+          "nextLink": null
+        }
+      }
+    }
```
Remove the parenthesis accordingly
```suggestion
```

ansahdev replied:
Updated.

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/examples/GetDiskInspection.json:
```diff
@@ -0,0 +1,26 @@
+        "x-ms-request-id": "27b7c568-16ec-46f3-bcf1-5bea3f2529b1"
+      },
+      "body": {
+        "properties": {
```
We should not have this level. This is the keyword of the schema definition, which should not appear in the value of that schema
```suggestion
```

ansahdev replied:
Updated

ansahdev committed:
1) Updated examples json based on requested change
2) Added "description" in definiton

ansahdev committed:
1) Updated examples json based on requested change
2) Added "description" in definiton

ansahdev committed:
1) Updated examples json based on requested change
2) Added "description" in definiton

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,377 @@
+            "type": "string"
+          },
+          "description": "Describes what are the supported resource types for a diagnostic",
+          "x-ms-examples": [
+            "Microsoft.Compute/virtualMachines"
+          ]
```
Please remove this - `x-ms-examples` can only exist in operation section not in `definitions`

ansahdev replied:
Removed the property.

ansahdev committed:
Removed "x-ms-examples" from "definitions" section to avoid LintDiff error.

ansahdev committed:
Removed "x-ms-examples" from "definitions" section to avoid LintDiff error.

ansahdev committed:
Removed "x-ms-examples" from "definitions" section to avoid LintDiff error.

ArcturusZhang commented on specification/compute/resource-manager/readme.md:
````diff
@@ -829,6 +829,23 @@ input-file:
 - Microsoft.ContainerService/preview/2015-11-01-preview/containerService.json
 ```
 
+### Tag: package-2021-06-01-preview
+
+These settings apply only when `--tag=2021-06-01-preview` is specified on the command line.
+
+``` yaml $(tag) == 'package-2021-06-01-preview'
+input-file:
+- Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json
+```
````
We need to include other compute swaggers in this tag too.

ansahdev replied:
Added swaggers of package 2021-03-01 under tag-package-2021-06-01-preview

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,374 @@
+      "required": true,
+      "type": "string",
+      "description": "The location used to execute the diagnostic operation.",
+      "pattern": "^[-\\w\\._]+$"
```
```suggestion
      "pattern": "^[-\\w\\._]+$",
      "x-ms-parameter-location": "method"
```

ansahdev replied:
Updated

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,374 @@
+            "schema": {
+              "$ref": "#/definitions/ComputeDiagnosticsList"
+            }
+          }
```
we need to add a default response here like the operations above (for instance line 121)

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,374 @@
+            "schema": {
+              "$ref": "#/definitions/ComputeDiagnosticBase"
+            }
+          }
```
Please add a default response

ansahdev replied:
Updated

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,374 @@
+  },
+  "definitions": {
+    "ComputeDiagnosticsList": {
+      "description": "Lists all available Compute diagnostics for a subscription in a location.",
+      "properties": {
```
```suggestion
      "description": "Lists all available Compute diagnostics for a subscription in a location.",
      "type": "object",
      "properties": {
```

ansahdev replied:
Updated

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,374 @@
+      }
+    },
+    "DiagnosticProperties": {
+      "description": "Contains additional properties of a diagnostic",
+      "properties": {
```
```suggestion
      "description": "Contains additional properties of a diagnostic",
      "type": "object",
      "properties": {
```

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,374 @@
+        }
+      }
+    },
+    "RunDiskInspectionInput": {
+      "properties": {
```
```suggestion
    "RunDiskInspectionInput": {
      "type": "object",
      "properties": {
```

ansahdev replied:
Updated

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,374 @@
+      ],
+      "description": "Data used for requesting a Disk Inspection execution."
+    },
+    "RunDiskInspectionAsyncOperationResult": {
+      "properties": {
```
```suggestion
    "RunDiskInspectionAsyncOperationResult": {
      "type": "object",
      "properties": {
```

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,374 @@
+      },
+      "description": "Api output result when disk inspection result is completed."
+    },
+    "ErrorResponse": {
+      "x-ms-external": true,
```
```suggestion
    "ErrorResponse": {
      "type": "object",
      "x-ms-external": true,
```

ansahdev replied:
Udpated

ArcturusZhang commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,374 @@
+      },
+      "description": "An error response from the Compute Diagnostic Resource Provider service."
+    },
+    "ErrorDetail": {
+      "properties": {
```
```suggestion
    "ErrorDetail": {
      "type": "object",
      "properties": {
```

ansahdev replied:
Updated

ArcturusZhang reviewed (changes requested):
Please resolve the LintDiff errors

ansahdev committed:
1) Updated diagnostic.json to fix Lint Diff errors
2) Updated readme to include all relevant swaggers.

ansahdev committed:
1) Updated diagnostic.json to fix Lint Diff errors
2) Updated readme to include all relevant swaggers.

ansahdev committed:
1) Updated diagnostic.json to fix Lint Diff errors
2) Updated readme to include all relevant swaggers.

ArcturusZhang committed:
fix model collision issue

ArcturusZhang committed:
fix model collision issue

ArcturusZhang committed:
fix model collision issue

ArcturusZhang commented:
/azp run

ArcturusZhang committed:
more fix

ArcturusZhang committed:
more fix

ArcturusZhang committed:
more fix

haagha commented on specification/compute/resource-manager/readme.md:
````diff
@@ -829,6 +829,30 @@ input-file:
 - Microsoft.ContainerService/preview/2015-11-01-preview/containerService.json
 ```
 
````
Hi @ansahdev did you intentionally add tag description at the bottom? The tag descriptions are listed in descending order and this should be at the top.

ansahdev replied:
Moved the new tags to the top of the list.

ansahdev committed:
Moved the new package tags "package-2021-06-01-preview" and "package-2021-06-01-preview-only" to the top of the list.

ansahdev committed:
Moved the new package tags "package-2021-06-01-preview" and "package-2021-06-01-preview-only" to the top of the list.

ansahdev committed:
Moved the new package tags "package-2021-06-01-preview" and "package-2021-06-01-preview-only" to the top of the list.

haagha commented on specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json:
```diff
@@ -0,0 +1,394 @@
+    },
+    "ErrorResponse": {
+      "type": "object",
+      "x-ms-external": true,
```
Hi @ansahdev Can you clarify why this is marked as x-ms-external:true? This prevents the object from being generated in the SDK which is resulting in SDK solution errors.

ansahdev replied:
I could have referred this from an existing RP json. What is the mitigation for the errors? Should this property be removed or updated with some additional information? Please advise.

ansahdev replied:
Removed the property.

ansahdev committed:
Removed "x-ms-external" : true to fix SDK generation errors

ansahdev committed:
Removed "x-ms-external" : true to fix SDK generation errors

ansahdev committed:
Removed "x-ms-external" : true to fix SDK generation errors

ansahdev committed:
Merge with master

ansahdev committed:
Merge from main

ansahdev committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into feature/cdrp-2021-06-01-preview

ansahdev commented:
Closing this as created a new PR with changes

JackTn commented:
Hi, @ansahdev. The PR has be closed for a long time and it's related branch still exist. Please tell me if you still need this branch or i will delete it in 14 days.
